#OpenXT:Writing connection settings to database to avoid creating them everytime on boot.
#       Instead of creating, the connections are read from database on boot by network-slave.
--- a/src/settings/plugins/keyfile/nms-keyfile-utils.c
+++ b/src/settings/plugins/keyfile/nms-keyfile-utils.c
@@ -24,6 +24,8 @@
 
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <errno.h>
+#include <sys/wait.h>
 
 #include "nm-keyfile-internal.h"
 #include "nm-utils.h"
@@ -34,6 +36,26 @@
 
 /*****************************************************************************/
 
+void
+xc_connections_update (char *path)
+{
+	char *argv[6] = { "/usr/bin/nm_sync.sh", "-n", "nm-connections", "-w", path, NULL };
+	int err, stat_loc;
+
+	err = fork();
+	if (err == -1) {
+		g_warning("%s.%d: cannot fork to nm_sync.sh: %d", __FILE__, __LINE__, errno);
+	} else if (err) {
+		while ((waitpid(err, &stat_loc, 0) == -1) && (errno == EINTR))
+			;
+		err = WEXITSTATUS(stat_loc);
+		if (err)
+			g_warning("%s.%d: nm_sync.sh failed with code: %d", __FILE__, __LINE__, err);
+	} else {
+		_exit(execvp(argv[0], argv));
+	}
+}
+
 char *
 nms_keyfile_loaded_uuid_filename (const char *dirname,
                                   const char *uuid,
--- a/src/settings/plugins/keyfile/nms-keyfile-utils.h
+++ b/src/settings/plugins/keyfile/nms-keyfile-utils.h
@@ -36,6 +36,8 @@ typedef enum {
 
 const char *nms_keyfile_utils_get_path (void);
 
+void xc_connections_update (char *path);
+
 /*****************************************************************************/
 
 char *nms_keyfile_loaded_uuid_filename (const char *dirname,
--- a/src/settings/nm-settings-connection.c
+++ b/src/settings/nm-settings-connection.c
@@ -612,6 +612,7 @@ nm_settings_connection_delete (NMSettingsConnection *self,
 	nm_settings_delete_connection (NM_SETTINGS_CONNECTION_GET_PRIVATE (self)->settings,
 	                               self,
 	                               allow_add_to_no_auto_default);
+	xc_connections_update (nm_settings_connection_get_filename (self));
 }
 
 /*****************************************************************************/
--- a/src/settings/plugins/keyfile/nms-keyfile-writer.c
+++ b/src/settings/plugins/keyfile/nms-keyfile-writer.c
@@ -341,6 +341,8 @@ _internal_write_connection (NMConnection
	
 	NM_SET_OUT (out_reread, g_steal_pointer (&reread));
	NM_SET_OUT (out_reread_same, reread_same);
+	xc_connections_update (path);
+
 	NM_SET_OUT (out_path, g_steal_pointer (&path));
 
 	return TRUE;
